File: Part 2 - Regression/Simple Linear Regression/simple_linear_regresion.R
# Regresion lineal simple

#importando dataset
dataset = read.csv('Salary_Data.csv')

#Dividir data set entre conjunto de entranamiento y testing
library(caTools)
set.seed(123) 
split = sample.split(dataset$Salary,SplitRatio = 2/3)
#Separando datos de entrenamiento y testing
training_set = subset(dataset,split==TRUE)
testing_set = subset(dataset,split==FALSE)

#Ajustar el modelo de regresion lineal simple con el conjunto de entrenamiento
#Variable dependiente en funcion(~) a la independiente
regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)
summary(regressor)

#Predecir resultados con el conjunto de tests
#Funcion generica. Indicar modelo y data (data con el minos nombre de columnas)
y_pred = predict(regressor, newdata=testing_set)
y_pred

library(ggplot2)
ggplot() + 
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             colour = "red") +#colocando puntos
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata=training_set)),
            colour = "blue") + #Colocando recta de regresion
  ggtitle("Sueldo vs Años de Experiencia (Conjunto de Entrenamiento)")+
  xlab("Años de Experiencia") + 
  ylab("Sueldo (en $)")

#Visualizacion de los resultados en el conjunto de testing
ggplot() + 
  geom_point(aes(x = testing_set$YearsExperience, y = testing_set$Salary),
             colour = "red") +#colocando puntos
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata=training_set)),
            colour = "blue") + #Colocando recta de regresion
  ggtitle("Sueldo vs Años de Experiencia (Conjunto de Testing)")+
  xlab("Años de Experiencia") + 
  ylab("Sueldo (en $)")

File: Part 2 - Regression/Simple Linear Regression/simple_linear_regresion.py
# -*- coding: utf-8 -*-
"""
Created on Tue Mar 28 19:39:27 2023

@author: josue
"""

#Regresion Lineal Simple

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

from sklearn.model_selection import train_test_split
#from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LinearRegression

dataset = pd.read_csv('Salary_Data.csv')

# matriz. Obteniendo valores del data frame [filas,columnas]
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

#Dividir data set entre conjunto de entranamiento y testing
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size = 1/3, random_state=0)

# sc_X = StandardScaler()
# X_train = sc_X.fit_transform(X_train)
# X_test = sc_X.transform(X_test)

#Creacion modelo de Regresion Lineal Simple - Metodo de los minimos cuadrados
regression = LinearRegression()
regression.fit(X_train, y_train)#Mismo tamaño
print("Constante/Intercepto:", regression.intercept_)
print("Pendiente:", regression.coef_)
#Predecir el conjunto de test
y_pred = regression.predict(X_test)

#Visualizar los resultados de entrenamiento
plt.scatter(X_train,y_train,color="red")#Nube de puntos
plt.plot(X_train, regression.predict(X_train))#Recta de regresion
plt.title("Sueldo vs Años de Experiencia (Conjunto de Entrenamiento)")
plt.xlabel("Años de experiencia")
plt.ylabel("Sueldo (en $)")

#Visualizar los resultados de testing
plt.scatter(X_test,y_test,color="purple")#Nube de puntos
plt.plot(X_train, regression.predict(X_train))#Recta de regresion
plt.title("Sueldo vs Años de Experiencia (Conjunto de Testing)")
plt.xlabel("Años de experiencia")
plt.ylabel("Sueldo (en $)")
